augmentation/augmentation.hpp

```diff
@@ -0,0 +1,174 @@
+ *
+ * @code
+ * Augmentation<> augmentation({"horizontal-flip", "resize = (224, 224)"}, 0.2);
+ * augmentation.Transform(dataloader.TrainFeatures);
+ * @endcode
+ * 
+ * @tparam DatasetType Datatype on which augmentation will be done.
+ */
+template<typename DatasetType = arma::mat>
+class Augmentation
+{
+ public:
+  //! Create the augmenation class object.
+  Augmentation();
+
+  /**
+   * Constructor for augmentation class.
+   *
+   * @param augmentations List of strings containing one of the supported
+   *                     augmentation.
+   * @param augmentationProbability Probability of applying augmentation on
+   *                                the dataset.
+   *                                NOTE : This doesn't apply to augmentations
+   *                                such as resize.
+   */
+  Augmentation(const std::vector<std::string>& augmentations,
+               const double augmentationProbability);
+
+  /**
+   * Applies augmentation to the passed dataset.
+   *
+   * @param dataset Dataset on which augmentation will be applied.
+   * @param datapointWidth Width of a single data point i.e.
+   *                    Since each column represents a seperate data
+   *                    point.
+   * @param datapointHeight Height of a single data point.
+   * @param datapointDepth Depth of a single data point. For 2-dimensional
+   *                     data point, set it to 1. Defaults to 1.
+   */
+  void Transform(DatasetType& dataset,
+                 const size_t datapointWidth,
+                 const size_t datapointHeight,
+                 const size_t datapointDepth = 1);
+
+  /**
+   * Applies resize transform to the entire dataset.
+   *
+   * @param dataset Dataset on which augmentation will be applied.
+   * @param datapointWidth Width of a single data point i.e.
+   *                    Since each column represents a seperate data
+   *                    point.
+   * @param datapointHeight Height of a single data point.
+   * @param datapointDepth Depth of a single data point. For 2-dimensional
+   *                     data point, set it to 1. Defaults to 1.
+   * @param augmentation String containing the transform.
+   */
+  void ResizeTransform(DatasetType& dataset,
+                       const size_t datapointWidth,
+                       const size_t datapointHeight,
+                       const size_t datapointDepth,
+                       const std::string& augmentation);
+
+ private:
+  /**
+   * Initializes augmentation map for the class.
+   */
+  void InitializeAugmentationMap();
+
+  /**
+   * Function to determine if augmentation has Resize function.
+   * @param augmentation Optional argument to check if a string has
+   *                     resize substring.
+   */
+  bool HasResizeParam(const std::string& augmentation = "")
+  {
+    if (augmentation.length())
+    {
+      return augmentation.find("resize") != std::string::npos;
+    }
```

Comment:
According the [Brace Placement in DesignGuidelines](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#brace-placement), let's remove the braces :)
```suggestion
    if (augmentation.length())
      return augmentation.find("resize") != std::string::npos;
```